pkgs/by-name/pa/packet/package.nix

```
@@ -0,0 +1,75 @@
+{
+  lib,
+  stdenv,
+  fetchFromGitHub,
+  cargo,
+  meson,
+  ninja,
+  pkg-config,
+  protobuf,
+  rustPlatform,
+  rustc,
+  wrapGAppsHook4,
+  cairo,
+  dbus,
+  gdk-pixbuf,
+  glib,
+  gtk4,
+  libadwaita,
+  pango,
+  blueprint-compiler,
+  desktop-file-utils,
+}:
+stdenv.mkDerivation (finalAttrs: {
+  pname = "packet";
+  version = "0.3.1";
+
+  src = fetchFromGitHub {
+    owner = "nozwock";
+    repo = "packet";
+    rev = finalAttrs.version;
+    hash = "sha256-Bk2J38NNrfixkYMveNQb9KkNG/pF3V9LgD1YGgmxg1c=";
+  };
+
+  cargoDeps = rustPlatform.importCargoLock {
+    lockFile = "${finalAttrs.src}/Cargo.lock";
+    outputHashes = {
+      "mdns-sd-0.10.4" = "sha256-y8pHtG7JCJvmWCDlWuJWJDbCGOheD4PN+WmOxnakbE4=";
+      "rqs_lib-0.11.5" = "sha256-wD7ZQ5OvIrjNHy3f3ss3bgNJn3l+7cUJjokonCLGE9w=";
+      "sys_metrics-0.2.7" = "sha256-hCialGTBTA7QdiqlTV6sBdiXpgDEB+96IWLFV2M+36o=";
+    };
+  };
```

This doesn't work in Nixpkgs because we don't allow something known as "import-from-derivation" (IFD). Simply put, the way `importCargoLock` works is that Nix has to parse the file and then produce the required Cargo dependencies, but to do that it has to build the source first, which means that one package would rely on the contents of another package and massively slow down evaluation times. You can see that from the failing CI and by running `nix-build -A packet --option allow-import-from-derivation false`.

Why not just `fetchCargoVendor`? You wouldn't even need to specify Git hashes

```suggestion
  cargoDeps = rustPlatform.fetchCargoVendor {
    inherit (finalAttrs) src;
    hash = "..."; 
  };
```